(#28)feat/map

nabitang12 opened the pull request:
## 📢 기능 설명

필요시 실행결과 스크린샷 첨부

https://github.com/user-attachments/assets/d066eff1-9982-4888-ae4d-5eab9d08fc3d

지도 마커 api 구현 (API 구현이 완료되면 하겠습니다..)
스피너 구현을 하긴했는데 로딩이 다 완료 되고도 아직 로딩완료창이 아주 짧게 보이는데 다시 고쳐보겠습니다 
컴포넌트 꾸미기 (검색 바 할인 filter 바 마커 애니메이션 구현)
마커 클릭시 가게 상세 페이지로 이동 (임시)
주석은 API 구현이 완료되면 삭제하겠습니다
## 연결된 issue

연결된 issue를 자동으로 닫기 위해 아래 {이슈넘버}를 입력해주세요. <br>
close #28 
<br>
<br>

## 🩷 Approve 하기 전 확인해주세요!

- [ ] 리뷰어가 확인해줬으면 하는 사항 적어주세요.
- [ ]

<br>

## ✅ 체크리스트

- [v]  PR 제목 규칙 잘 지켰는가?
- [v] 추가/수정사항을 설명하였는가?
- [v] 이슈넘버를 적었는가?
- [v] Approve 하기 전 확인 사항 체크했는가?

nabitang12 committed:
modified:   package.json
	modified:   src/components/MapCard/GoogleMapCard/Map.style.ts
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/components/MapCard/SearchCard/SearchBar.style.ts
	new file:   src/pages/MapPage/MapRender.style.ts
	modified:   src/pages/MapPage/MapRender.tsx

ssolfa commented:
그냥 열어두시고 커밋 다 하신 후에 리뷰어 체크해주세요!! 그럼 달려오겠습니다 🏃🏃🏃

nabitang12 committed:
remove .env file from git repo

nabitang12 committed:
Merge branch '(#33)feat/storeInfoStore' of https://github.com/UMC-6th-BAB/BAB-FE into (#5)design/map-page

nabitang12 committed:
searchBar style 수정, 임시 더미 데이터 커밋

nabitang12 committed:
더미데이터 이용해서 음식점 검색 시 마커 표시 구현

nabitang12 committed:
fix: 마커 검색 애니메이션 구현 반응형 searchBar 구현

nabitang12 committed:
fix: 마커 클릭시 가게 페이지로 이동 임시로 구현 로딩 스피너 재구현

nabitang12 committed:
feat: 마커 API 적용

nabitang12 committed:
fix: styledcomponent props 수정

Jayjunyoung commented on src/apis/Store/fetchMarker.ts:
```diff
@@ -0,0 +1,16 @@
+
+export const fetchMarker = async () => {
+  try {
+    const res = await axios.get(`https://babedeuk.shop:8080/v1/stores`, {
```
`const API_BASE_URL = import.meta.env.VITE_API_BASE_URL` 로 가져오신담에
`${API_BASE_URL}/v1/stores`

로 바꿔주세요!

Jayjunyoung commented on src/components/MapCard/GoogleMapCard/Map.tsx:
```diff
@@ -143,16 +140,30 @@ export default function Map({
-              : yellowIcon(info.menus[0].price, info.menus[0].discountPrice)
+            info.menu[0].discountPrice === 0
+              ? greyIcon(info.menu[0].price)
+              : yellowIcon(info.menu[0].price, info.menu[0].discountPrice)
```
여기는 아직 API연동 안돼서 임시로 하셨군요! 연동 이후에 response 보니까 menuPrice랑 discountPrice 넘어오더라구요! 그거 이용해서 렌더링 해주시면 됩니다

Jayjunyoung replied:
![image](https://github.com/user-attachments/assets/94097534-9ddb-4d27-a4ff-9c73fee434be)

Jayjunyoung commented on src/components/MapCard/GoogleMapCard/Map.tsx:
```diff
@@ -143,16 +140,30 @@ export default function Map({
           const markerView = new AdvancedMarkerElement({
             map: googleMap,
-            position: new google.maps.LatLng(info.lat, info.lng),
+            position: new google.maps.LatLng(info.lat as number, info.lng),
```
경도 위도 둘다 타입단언 number로 해주시거나 interface 만들어서 적용해주시는게 좋을 것 같습니다!

nabitang12 committed:
feat: 임시 API 구현

nabitang12 committed:
feat: 임시 API 검색, 초기 마커 설정, 클릭시 상세페이지 navigate 구현

nabitang12 committed:
Merge branch 'develop' into (#5)design/map-page

nabitang12 committed:
modified:   src/apis/Store/fetchMarker.ts
	modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/main.tsx
	modified:   src/pages/MapPage/MapRender.tsx
	modified:   src/stores/storeInfoStore.ts
	modified:   src/stores/tempStore.ts

nabitang12 committed:
modified:   src/components/MapCard/GoogleMapCard/Map.tsx
	modified:   src/pages/MapPage/MapRender.tsx

nabitang12 committed:
feat: 지도 API 구현

Jayjunyoung commented on src/apis/Marker/fetchMarker.ts:
```diff
@@ -0,0 +1,17 @@
+  try {
+    const res = await axios.get(`${baseUrl}/v1/stores`, {
+      headers: {
+        Authorization: `Bearer eyJhbGciOiJIUzI1NiJ9.eyJpZCI6Nywicm9sZSI6InN0dWRlbnQiLCJpYXQiOjE3MjM5NzY4NzksImV4cCI6MTcyNTE4NjQ3OX0.AV3UHwB_XGiMCttKhs7T-8GaVBGlnUWFE0q3WJ-aLVA`,
```
이 부분 token인거면 변수에 담은담에 `bearer ${token}`
이런식으로 바꿔주시면 좋을거 같애요!

Jayjunyoung replied:
토큰이 유출되어있어서 env에서 가져와서 token이라는 변수에 담고 적용시켜줘야할거 같습니다!

Jayjunyoung commented on src/apis/Marker/fetchSearchStore.ts:
```diff
@@ -0,0 +1,43 @@
+      {
+        headers: {
+          Authorization:
+            'Bearer eyJhbGciOiJIUzI1NiJ9.eyJpZCI6Nywicm9sZSI6InN0dWRlbnQiLCJpYXQiOjE3MjM5NzY4NzksImV4cCI6MTcyNTE4NjQ3OX0.AV3UHwB_XGiMCttKhs7T-8GaVBGlnUWFE0q3WJ-aLVA',
```
여기도 동일하게 부탁드립니다 !!

nabitang12 committed:
fix: token 값 env 파일에 설정

nabitang12 committed:
fix: API 수정 및 오류 수정

nabitang12 committed:
fix: 로직 오류 수정

nabitang12 committed:
fix: token 값 변경

yongmin01 reviewed (approved):
수고하셨습니다 ㅠㅠㅠㅠㅠ !!
